metadata-io/src/main/java/com/linkedin/metadata/entity/ebean/EbeanEntityService.java

```diff
@@ -614,21 +611,19 @@ public RollbackRunResult deleteUrn(Urn urn) {
     List<AspectRowSummary> removedAspects = new ArrayList<>();
     Integer rowsDeletedFromEntityDeletion = 0;
 
+    final EntitySpec spec = getEntityRegistry().getEntitySpec(urnToEntityName(urn));
+    final AspectSpec keySpec = spec.getKeyAspectSpec();
     String keyAspectName = getKeyAspectName(urn);
+
     EbeanAspectV2 latestKey = _entityDao.getLatestAspect(urn.toString(), keyAspectName);
     if (latestKey == null || latestKey.getSystemMetadata() == null) {
       return new RollbackRunResult(removedAspects, rowsDeletedFromEntityDeletion);
     }
 
     SystemMetadata latestKeySystemMetadata = parseSystemMetadata(latestKey.getSystemMetadata());
-
     RollbackResult result = deleteAspect(urn.toString(), keyAspectName, Collections.singletonMap("runId", latestKeySystemMetadata.getRunId()));
-    Optional<AspectSpec> aspectSpec = getAspectSpec(result.entityName, result.aspectName);
-    if (!aspectSpec.isPresent()) {
-      log.error("Issue while rolling back: unknown aspect {} for entity {}", result.entityName, result.aspectName);
-    }
```

Comment:
what is the significance of this change? I'm not entirely sure why it was added actually. i think this had to do w/ people loading in multiple entity registries

Reply:
We already have to retrieve the key aspect spec above, so now I'm just reusing it

Reply:
resolved offline- seems like this is not a problem since its already being fetched